### PR TITLE
Feature/04 pig latin

### PR DESCRIPTION
**Pull request by salcudeann**

### CODE REVIEW

File: 04_pig_latin/pig_latin.rb [hunk 1]
```diff
@@ -0,0 +1,33 @@
+$vow = ['a', 'e', 'i', 'o', 'u']
```
Comment (alex-damian-negru) [inline 04_pig_latin/pig_latin.rb]: Variables declared with `$` are global. Don't use global variables.

Comment (alex-damian-negru) [reply 04_pig_latin/pig_latin.rb]: `vow` is unclear, better use the whole, pluralized name, e.g. `vowels`.

File: 04_pig_latin/pig_latin.rb [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+$con = ['b', 'c', 'd', 'f', 'g', 'h', 'j', 'k', 'l', 'm', 'n', 'p', 'q', 'r', 's', 't', 'w', 'v', 'x', 'y', 'z']
```
Comment (alex-damian-negru) [inline 04_pig_latin/pig_latin.rb]: Same here.

File: 04_pig_latin/pig_latin.rb [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+
+def translate(str)
+  str.split.map { |curr_word|  translate_word(curr_word) }.join(' ')
```
Comment (alex-damian-negru) [inline 04_pig_latin/pig_latin.rb]: :+1:

File: 04_pig_latin/pig_latin.rb [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+end
+
+def translate_word(word)
+  final_word = ''
+  word[0] = word[0].downcase
+
+  if $vow.include? word[0]
```
Comment (alex-damian-negru) [inline 04_pig_latin/pig_latin.rb]: You an calso use [.start_with?](https://ruby-doc.org/core-3.0.2/String.html#method-i-start_with-3F).

File: 04_pig_latin/pig_latin.rb [hunk 5]
```diff
@@ -0,0 +1,33 @@
...
+    final_word += translate_vow(word)
```
Comment (alex-damian-negru) [inline 04_pig_latin/pig_latin.rb]: I think the `final_word` variable isn't needed - try to find a solution that doesn't use it.

File: 04_pig_latin/pig_latin.rb [hunk 6]
```diff
@@ -0,0 +1,33 @@
...
+
+def translate_con(word)
+  new_word = ''
+  i = 0
+  while $con.include? word[i]
+    if (word[i]=='q') && word[i+1]=='u'
+      i = i+1
+    end
+    i += 1
+  end
```
Comment (alex-damian-negru) [inline 04_pig_latin/pig_latin.rb]: We rarely use `while` in the Ruby world (I don't remember the last time I used it).

Can you find a solution that uses `each`, `map`, `split` (one or a combination of these)?

**Review by alex-damian-negru — CHANGES_REQUESTED** [review]

File: 04_pig_latin/pig_latin.rb [hunk 7]
```diff
@@ -0,0 +1,23 @@
...
+end
+
+def translate_word(word)
+  vowels = ['a', 'e', 'i', 'o', 'u']
+  con = ['b', 'c', 'd', 'f', 'g', 'h', 'j', 'k', 'l', 'm', 'n', 'p', 'q', 'r', 's', 't', 'w', 'v', 'x', 'y', 'z']
+  word[0] = word[0].downcase
+
+  if word[0].start_with?(*vowels)
+     word + 'ay'
+  else
```
Comment (pirvudoru) [inline 04_pig_latin/pig_latin.rb]: Can you rewrite this to not make the first letter an exceptional case?

File: 04_pig_latin/pig_latin.rb [hunk 8]
```diff
@@ -0,0 +1,22 @@
+def translate(str)
+  str.split.map { |curr_word| translate_word(curr_word) }.join(' ')
+end
+
+def translate_word(word)
+  cons = ['b', 'c', 'd', 'f', 'g', 'h', 'j', 'k', 'l', 'm', 'n', 'p', 'q', 'r', 's', 't', 'w', 'v', 'x', 'y', 'z']
```
Comment (pirvudoru) [inline 04_pig_latin/pig_latin.rb:6]: You could use 

```
vowels = ['a', 'e', 'i', 'o', 'u']
````

and 

```
de vowel?(char)
  vowels.inlcude(char)
end

def consonant?(char)
  !vowel?(char)
end
```

for a smaller array :))

**Review by pirvudoru — APPROVED** [review]

**Comment by alex-damian-negru** [comment]
@salcudeann Please merge the PRs only until after you've replied to or applied all the change suggestions here.

**Pull request merged**